sdk/src/main/java/com/qonversion/android/sdk/internal/QProductCenterManager.kt

```diff
@@ -981,6 +1037,10 @@ internal class QProductCenterManager internal constructor(
                     override fun onError(error: QonversionError) {
                         storeFailedPurchaseIfNecessary(purchase, purchaseInfo, product)
 
+                        product?.storeID?.let {
+                            removePurchaseOptions(it)
+                        }
```

Comment:
In two places we don't check the key for null before the call, here we do check - let's unify it. I think the `removePurchaseOptions` function shouldn't work with nulls, or at least check the key for null and only then call the nested function `update...`

Reply:
updatePurchaseOptions function can handle null store ids